packages/compass-databases-navigation/src/databases-navigation-tree.tsx

```diff
@@ -0,0 +1,744 @@
+    </li>
+  );
+};
+
+const CollectionsListPlaceholder: React.FunctionComponent<{
+  length: number;
+  isVisible: boolean;
+}> = ({ length, isVisible }) => {
+  const items = useMemo(() => {
+    return Array.from({ length }, (_, index) => (
+      <PlaceholderItem key={index}></PlaceholderItem>
+    ));
+  }, [length]);
+
+  return (
+    <ul
+      role="presentation"
+      data-testid="placeholder"
+      className={cx(
+        ulReset,
+        placeholderContainer,
+        !isVisible && placeholderContainerHidden
+      )}
+    >
+      {items}
+    </ul>
+  );
+};
+
+const actionsContainer = css({
+  flex: 'none',
+  marginLeft: 'auto',
+  alignItems: 'center',
+  display: 'flex',
+});
+
+const databaseItem = css({
+  height: DATABASE_ROW_HEIGHT,
+  paddingLeft: spacing[1],
+  paddingRight: spacing[1],
+  display: 'flex',
+  alignItems: 'center',
+  backgroundColor: backgroundColor,
+});
+
+const activeBackground = css({
+  backgroundColor: backgroundColorActive,
+});
+
+const databaseItemContainer = css({
+  contentVisibility: 'auto',
+  overflow: 'hidden',
+  outline: 'none',
+});
+
+const buttonReset = css({
+  padding: 0,
+  margin: 0,
+  background: 'none',
+  border: 'none',
+});
+
+const databaseIconContainer = css({
+  flex: 'none',
+  fontSize: 0,
+  // Not using leafygreen spacing here because none of them allow to align the
+  // button with the search bar content. This probably can go away when we are
+  // rebuilding the search also
+  padding: 6,
+  transition: 'transform .16s linear',
+  transform: 'rotate(0deg)',
+});
+
+const iconRotated = css({
+  transform: 'rotate(90deg)',
+});
+
+const databaseItemName = css({
+  overflow: 'hidden',
+  whiteSpace: 'nowrap',
+  textOverflow: 'ellipsis',
+  marginLeft: spacing[1],
+});
+
+// Using important here because leafygreen / emotion applies styles in the order
+// that doesn't allow our styles override theirs
+const iconButtonSmall = css({
+  width: `${spacing[4]}px !important`,
+  height: `${spacing[4]}px !important`,
+});
+
+const iconButtonSmallActive = css({
+  color: 'currentColor !important',
+});
+
+const IconButtonSmall: React.FunctionComponent<{
+  glyph: string;
+  label: string;
+  title?: string;
+  onClick(evt: React.MouseEvent<HTMLButtonElement>): void;
+  isActive: boolean;
+  'data-testid'?: string;
+}> = ({ glyph, label, onClick, isActive, children, title, ...rest }) => {
+  return (
+    <IconButton
+      className={cx(iconButtonSmall, isActive && iconButtonSmallActive)}
+      aria-label={label}
+      title={title}
+      onClick={onClick}
+      darkMode
+      data-testid={rest['data-testid']}
+    >
+      <Icon role="presentation" glyph={glyph} size="small"></Icon>
+      {/* Only here to make leafygreen menus work */}
+      {children}
+    </IconButton>
+  );
+};
+
+const collectionsListContainer = css({
+  position: 'relative',
+});
+
+const DatabaseItem: React.FunctionComponent<
```

Comment:
So the reason I kept them in one file is that they are not used outside and they do have overlapping styles that are shared, I was trying to avoid having e.g., a five line file for `actionsContainer` styles that are shared between databases and collections, for me personally it creates a ton of load if I need to jump to a file just to see these. As I mentioned in the PR description I don't mind splitting them though if this feels like too much for one

Reply:
That's fair, I think a few of the styles, like the focus ring, could be something we share with the connections sidebar. Fine as it is, but if we find we're adding more logic here I think it's definitely worth breaking into more files.